twilight-http/src/request/guild/auto_moderation/create_auto_moderation_rule.rs

```diff
@@ -198,9 +206,16 @@ impl<'a> CreateAutoModerationRule<'a> {
 
     /// Append an action of type [`Timeout`].
     ///
+    /// # Errors
+    ///
+    /// Returns [`AutoModerationActionMetadataDurationSeconds`] if the duration
+    /// is invalid.
+    ///
     /// [`Timeout`]: AutoModerationActionType::Timeout
+    /// [`AutoModerationActionMetadataDurationSeconds`]: twilight_validate::request::ValidationErrorType::AutoModerationActionMetadataDurationSeconds
     pub fn action_timeout(mut self, duration_seconds: u32) -> Self {
```

Comment:
Should I write it like this?
Throws [`AutoModerationActionMetadataDurationSeconds`] when executing the request if the `duration_seconds` field is invalid.

Reply:
i think moving it to the method that returns the error is the idiomatic approach, but why dont we just return the error right here? like we do in `CreateMessage::content` for example

Reply:
The error handling was changed in [d0876f7](https://github.com/twilight-rs/twilight/commit/d0876f7a9a01fcb0a62684ccd6d2200bdb707cf3).  Also `CreateMessage::content` uses my current approach as I saw [here](https://github.com/twilight-rs/twilight/blob/next/twilight-http/src/request/channel/message/create_message.rs#L173-L192).

Reply:
oh i see resolving it then thank you

Reply:
actually do we want to either rename the error types to better tell that its an error type the way `ContentInvalid` isnt named `Content` or quality the path to `ValidationErrorType::AutoModerationActionMetadataDurationSeconds`

Reply:
I already think they are way too long and currently they are [consistent](https://github.com/twilight-rs/twilight/blob/next/twilight-validate/src/request.rs#L394-L525) with the other Errors in the file.

Reply:
then qualifying it would make it more clear

Reply:
this was missed while qualifying

Reply:
this is a outdated file

Reply:
This is inside the validate crate and not the automoderation interface. Changing this would be inconsistent with the rest of the validation functions in this file. Should I really change this?

Reply:
ahh sorry my bad i missed that part, thank you for the pr